.github/workflows/test.yml

```diff
@@ -240,47 +240,49 @@ jobs:
           sudo apt update -y
           sudo apt install -y --no-install-recommends intel-oneapi-runtime-compilers intel-oneapi-runtime-opencl
       - name: Install packages (cuda)
-        if: matrix.backend == 'cuda' || matrix.backend == 'ptx'
+        if: matrix.backend == 'cuda' || matrix.backend == 'ptx' || matrix.backend == 'triton'
         run: |
           sudo apt update -y
           sudo apt install -y --no-install-recommends git g++ cmake ninja-build llvm-15-dev zlib1g-dev libglew-dev flex bison libfl-dev libboost-thread-dev libboost-filesystem-dev nvidia-cuda-toolkit-gcc
       - name: Cache gpuocelot
-        if: matrix.backend == 'cuda' || matrix.backend == 'ptx'
+        if: matrix.backend == 'cuda' || matrix.backend == 'ptx'|| matrix.backend == 'triton'
         id: cache-build
         uses: actions/cache@v3
         env:
           cache-name: cache-gpuocelot-build
         with:
           path: ${{ github.workspace }}/gpuocelot/ocelot
-          key: ubuntu22.04-gpuocelot-19626fc00b6ee321638c3111074269c69050e091
+          key: ubuntu22.04-gpuocelot-szymonozog-tinygrad
       - name: Clone/compile gpuocelot
-        if: (matrix.backend == 'cuda' || matrix.backend == 'ptx') && steps.cache-build.outputs.cache-hit != 'true'
+        if: (matrix.backend == 'cuda' || matrix.backend == 'ptx' || matrix.backend == 'triton') && steps.cache-build.outputs.cache-hit != 'true'
         run: |
-          git clone --recurse-submodules https://github.com/gpuocelot/gpuocelot.git ${{ github.workspace }}/gpuocelot
+          git clone --recurse-submodules --single-branch --branch tinygrad https://github.com/SzymonOzog/gpuocelot.git ${{ github.workspace }}/gpuocelot
           cd ${{ github.workspace }}/gpuocelot/ocelot
-          git checkout 19626fc00b6ee321638c3111074269c69050e091
```

Comment:
How is the upstreaming going? I like that this pins a commit

Reply:
the fixes still haven't been merged nor reviewed

Reply:
Sorry for delay, I've merged them now. Somehow, notifications miss my attention. Your work is very valuabe, so next time, if I miss something, please feel free to ping me by mentioning @dmikushin or a personal message.

Reply:
Sorry for delay, I've merged them now. Somehow, notifications miss my attention. Your work is very valuabe, so next time, if I miss something, please feel free to ping me by mentioning @dmikushin or a personal message.

Reply:
Thanks a lot for the merge, and the heads up. I definetly won't hesitate to ping in the future if we'll need a fast response